### ISSUE TITLE
Search-query formulation in search edit box requiring certain sequence of search terms

### ISSUE THREAD

**Issue by pdeschner** [harvard-lil/nuremberg#14]
Of the 2 following seemingly identical queries, only 1. works:
1. exhibit type:transcript
2. type:transcript exhibit

Can search be fixed to allow any sequence of terms?

**Comment by emmalemma** [comment]
This was by design, as I wanted to allow for queries such as `date:January 2`. So the design was that only the initial part of the query is free keyword search, while field queries consume the rest of the keywords.

I still think that's a bit simpler, but I can understand the alternative interpretation. So I've changed the query parsing to work as you expected, with field queries consuming only a single keyword, such that these are equivalent:

```
exhibit type:transcript
type:transcript exhibit
```

But to preserve the old behavior I added a syntax to search for multiple keywords, so these are equivalent:

```
date:January date:2
date:(January 2)
date:(2 January)
```

This is still distinct from the exact phrase search, as in: `date:"January 2"`, which will not find any results.

Let me know if this seems better, and if you notice any issues with the new parsing.
